### DISCUSSION TITLE
Custom random

### DISCUSSION THREAD

**Discussion by aarondcohen** [aarondcohen/id128#8] [Ideas]
It may be useful to let developers provide their own random byte generator.  I could see cases where someone may want to use a high quality pseudorandom number generator or a generator that allocates additional randomness in the background so that the primary thread doesn't lock up.

Feedback from developers would be useful.

**Comment by jvliwanag** [comment]
Another use case for this on react native. While there's no full `crypto` implementation on react-native, there's enough functionality to create secure random bytes.
